### ISSUE TITLE
How to detect targets with an aspect ratio greater than 100?

### ISSUE THREAD

**Issue by yuanshuai666** [ultralytics/ultralytics#18803]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

When I was training with YOLOv8, I found that it could not detect targets with an aspect ratio greater than 100. Are there any parameter limitations or solutions for this issue?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @yuanshuai666, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. Be sure to include a clear description of the issue, setup details, and relevant code or data snippets to allow us to reproduce it effectively.

If this is a custom training ❓ Question, please provide as much information as possible, such as:

- The specifics of your dataset (examples of images/labels)
- Training configurations (hyperparameters, model size, epochs, augmentation settings, etc.)
- Validation results or logs
- Any relevant code snippets used for training/evaluation

Additionally, you may find our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) extremely helpful for optimizing your model's performance.

Join the Ultralytics community where it suits you best. For real-time discussions, check out [Discord](https://discord.com/invite/ultralytics) 🎧. For more in-depth topics, visit [Discourse](https://community.ultralytics.com/), or discuss with others on our [Subreddit](https://reddit.com/r/Ultralytics) 🐙.

## Upgrade

Please ensure you are using the latest `ultralytics` version to rule out any resolved issues. Upgrade in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) by running:

```bash
pip install -U ultralytics
```

## Environments

You can run YOLOv8 in various verified environments where dependencies like [CUDA](https://developer.nvidia.com/cuda-zone), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) are preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests confirm the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to guide you toward helpful resources 🌟. An Ultralytics engineer will review your question and provide further assistance soon.

**Comment by Y-T-G** [comment]
If the objects are really long, then the receptive field of YOLO's anchors aren't large enough to see the whole object.

**Comment by yuanshuai666** [comment]
> If the objects are really long, then the receptive field of YOLO's anchors aren't large enough to see the whole object.

Thanks for your reply! As far as I know, YOLOv8 is anchor-free, but does it have a design similar to anchors? If so, how can I adjust the anchor? If not, how can I adjust it to enable YOLOv8 to detect such targets? Or do other versions of YOLO, such as YOLOv11, have related improvements?

**Comment by Y-T-G** [comment]
It's anchor-free in the sense that it doesn't use predefined anchor boxes. But it instead uses the grid-cells as anchor points like FCOS. The grid-cells have limited receptive field depending on the scale.

**Comment by yuanshuai666** [comment]
> It's anchor-free in the sense that it doesn't use predefined anchor boxes. But it instead uses the grid-cells as anchor points like FCOS. The grid-cells have limited receptive field depending on the scale.

Thank you for your response. How can I expand the receptive field to detect such targets? Where do I need to modify? Could you provide example code?

**Comment by Y-T-G** [comment]
You need to add more depth.

Like this has an extra scale called P6:
https://github.com/ultralytics/ultralytics/blob/5bdcf9116d6284f24b0ad303cc921eccecff8c24/ultralytics/cfg/models/v8/yolov8-p6.yaml

This is the original:
https://github.com/ultralytics/ultralytics/blob/5bdcf9116d6284f24b0ad303cc921eccecff8c24/ultralytics/cfg/models/v8/yolov8.yaml

**Comment by Y-T-G** [comment]
How large are your objects in pixels?

**Comment by yuanshuai666** [comment]
> How large are your objects in pixels?

My original image size is 1440x100, and the target size can be referenced from the image below：

![Image](https://github.com/user-attachments/assets/a2f138e8-5511-4005-95eb-6bd29ca9ea55)
![Image](https://github.com/user-attachments/assets/2ed05164-b127-4e63-8684-8bb9cfcd1c43)

Also, how can I reference the P6 model in the training code? Could you provide an example?

**Comment by glenn-jocher** [comment]
Thank you for the clarification. For targets with such extreme aspect ratios, utilizing the P6 model is a good approach as it introduces an additional scale level, improving detection for larger or elongated objects. Here's how you can reference the P6 model in your training:

```bash
yolo train model=yolov8-p6.yaml data=your_dataset.yaml epochs=100 imgsz=1440
```

Replace `your_dataset.yaml` with your dataset configuration. Adjust `imgsz` to match your image dimensions for optimal results. Let me know if you need further assistance!

**Comment by yuanshuai666** [comment]
> Thank you for the clarification. For targets with such extreme aspect ratios, utilizing the P6 model is a good approach as it introduces an additional scale level, improving detection for larger or elongated objects. Here's how you can reference the P6 model in your training:
> 
> yolo train model=yolov8-p6.yaml data=your_dataset.yaml epochs=100 imgsz=1440
> Replace `your_dataset.yaml` with your dataset configuration. Adjust `imgsz` to match your image dimensions for optimal results. Let me know if you need further assistance!

Thank you very much for your help. I tried training with the P6 model, but it still cannot recognize long and slender targets with large aspect ratios. From the training results, it can be observed that many slender targets are only partially detected, and there is overlapping, making it impossible to recognize the entire outline of the target. Below is one of the results:

![Image](https://github.com/user-attachments/assets/30cf78f3-4929-4787-8eb7-e45f781fe0a4)

Below is my training code. Could you give me some more advice?

``from ultralytics import YOLO

model = YOLO("yolov8-p6.yaml").load("yolov8x.pt")
results = model.train(
    data='D:/code/Detection/datasets/dataset.yaml',  # data.yaml 文件的路径
    epochs=100,
    imgsz=1440, 
    batch=24,
    lr0=0.005,
    lrf=0.1,
    weight_decay=0.001,
    rect=True,
    device=[0]
)``

**Comment by cqjnovo** [comment]
> > How large are your objects in pixels?你的物体像素大小是多少？
> 
> My original image size is 1440x100, and the target size can be referenced from the image below：我的原始图片尺寸是 1440x100，目标尺寸可以参考下面的图片：
> 
> <img alt="Image" width="822" height="57.078125" src="https://private-user-images.githubusercontent.com/33718549/405679930-a2f138e8-5511-4005-95eb-6bd29ca9ea55.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQzODc3NzYsIm5iZiI6MTc2NDM4NzQ3NiwicGF0aCI6Ii8zMzcxODU0OS80MDU2Nzk5MzAtYTJmMTM4ZTgtNTUxMS00MDA1LTk1ZWItNmJkMjljYTllYTU1LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTI5VDAzMzc1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY0M2QzZjUwYzg1ZWFjYTNhMjM0OTFjMDg0OTExZjA3MGQ2NDFhNWE0NjE3MDk2NzdkZDE0NDEyODUwN2I1YjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-GnnZybXnxIUcWN6JmOf5DqubjEVuLmAIDjb6k8LxH0"> <img alt="Image" width="822" height="57.078125" src="https://private-user-images.githubusercontent.com/33718549/405679929-2ed05164-b127-4e63-8684-8bb9cfcd1c43.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQzODc3NzYsIm5iZiI6MTc2NDM4NzQ3NiwicGF0aCI6Ii8zMzcxODU0OS80MDU2Nzk5MjktMmVkMDUxNjQtYjEyNy00ZTYzLTg2ODQtOGJiOWNmY2QxYzQzLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTI5VDAzMzc1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3MGM4OGRlNjYxNTBjMjQ2ZWY1Yjg2MWNkNWQ2MjY1NzQ3YjE3MjVmZDIxYzdjNmVmNTQ0OGY2YjBkN2ZjYTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.lVQB9klLhTerPhqGOHCUUvRs9UVqVTvdsFRdjpglr5A">
> 
> Also, how can I reference the P6 model in the training code? Could you provide an example?另外，我如何在训练代码中引用 P6 模型？你能举个例子吗？

对于极端细长的目标，gt框的高度太低，在进行候选anchor筛选时，p5阶段的anchor几乎不会落在gt框内，模型用于预测的anchor均为p3的anchor，即使改了模型结构使用p6的检测头还是会出现感受野不够的问题。通过修改tal中的正样本筛选策略，按照gt框中心半径范围内的p5anchor，可以缓解此问题，预测框的宽度可以正常预测，但高度还是学的不是那么好，仍然不够贴合。不知道您这边有没有新的发现，有没有更好的解决方案，我们可以加个微信讨论一下吗

**Comment by glenn-jocher** [comment]
Your TAL adjustment makes sense, but for objects this thin the main bottleneck is input resolution and box parameterization rather than just the feature level: in practice you usually need to either pre‑process so the object height is larger in network space (e.g. vertically pad/resize so the line is at least tens of pixels high) or model the two endpoints/ keypoints and draw the line in post‑processing, as suggested earlier; switching to YOLO11 or adding more FPN levels will not fully remove this limitation. We don’t provide support via WeChat, so please continue the discussion here (or open a new issue/discussion with a minimal reproducible example), and you can also refer to the [Ultralytics detection docs](https://docs.ultralytics.com/tasks/detect/) for additional guidance on handling extreme shapes.

**Comment by zhangzhenyu-zzy** [comment]
> > > 你的物体像素大小是多少？你的物体像素大小是多少？
> > 
> > 
> > 我的原始图片尺寸是 1440x100，目标尺寸可以参考下面的图片：我的原始图片尺寸是 1440x100，目标尺寸可以参考下面的图片：
> > <img alt="图片" width="822" height="57.078125" src="https://private-user-images.githubusercontent.com/33718549/405679930-a2f138e8-5511-4005-95eb-6bd29ca9ea55.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQzODc3NzYsIm5iZiI6MTc2NDM4NzQ3NiwicGF0aCI6Ii8zMzcxODU0OS80MDU2Nzk5MzAtYTJmMTM4ZTgtNTUxMS00MDA1LTk1ZWItNmJkMjljYTllYTU1LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTI5VDAzMzc1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY0M2QzZjUwYzg1ZWFjYTNhMjM0OTFjMDg0OTExZjA3MGQ2NDFhNWE0NjE3MDk2NzdkZDE0NDEyODUwN2I1YjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-GnnZybXnxIUcWN6JmOf5DqubjEVuLmAIDjb6k8LxH0"> <img alt="图片" width="822" height="57.078125" src="https://private-user-images.githubusercontent.com/33718549/405679929-2ed05164-b127-4e63-8684-8bb9cfcd1c43.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQzODc3NzYsIm5iZiI6MTc2NDM4NzQ3NiwicGF0aCI6Ii8zMzcxODU0OS80MDU2Nzk5MjktMmVkMDUxNjQtYjEyNy00ZTYzLTg2ODQtOGJiOWNmY2QxYzQzLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTI5VDAzMzc1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3MGM4OGRlNjYxNTBjMjQ2ZWY1Yjg2MWNkNWQ2MjY1NzQ3YjE3MjVmZDIxYzdjNmVmNTQ0OGY2YjBkN2ZjYTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.lVQB9klLhTerPhqGOHCUUvRs9UVqVTvdsFRdjpglr5A">
> > 另外，我如何在训练代码中引用P6模型？你能举个例子吗？另外，我如何在训练代码中引用 P6 模型？你能举个例子吗？
> 
> 对于极端细长的目标，gt框的高度太低，在进行候选anchor筛选时，p5阶段的anchor几乎不会落在gt框内，模型用于预测的anchor均为p3的anchor，即使改了模型结构使用p6的检测头还是会出现感受野不够的问题。通过修改tal中的正样本筛选策略，按照gt框中心半径范围内的p5anchor，可以缓解此问题，预测框的宽度可以正常预测，但高度还是学的不是那么好，仍然不够贴合。不知道您这边有没有新的发现，有没有更好的解决方案，我们可以加个微信讨论一下吗

您好，可以加您一个微信联系吗？我也遇到了相同的问题，我的微信是zzydddd

**Comment by glenn-jocher** [comment]
We can’t provide support via WeChat/DM (and it’s best not to post personal contact info publicly); please open a new thread in [Ultralytics Discussions](https://github.com/orgs/ultralytics/discussions) and include the details from the [Minimum Reproducible Example guide](https://docs.ultralytics.com/help/minimum-reproducible-example/) plus your environment info from: 

```bash
yolo checks
```

For 1440×100 images with extremely thin objects, box detection often fragments into multiple partial boxes; the most reliable workaround is to rotate/tiling-preprocess so the object becomes “thicker” in network pixels, or re-label as two endpoints (pose) and reconstruct the line in post-processing.
